src/mo_frontend/parser.mly

```diff
@@ -830,7 +830,10 @@ dec_var :
     { VarD(x, annot_exp e t) @? at $sloc }
 
 dec_nonvar :
-  | LET p=pat EQ e=exp(ob)
+  | LET p=pat EQ e=exp(ob) %prec LET_NO_ELSE
+    { let p', e' = normalize_let p e in
+      LetD (p', e') @? at $sloc }
+  | LET p=pat EQ e=exp(ob) ELSE fail=exp(bl)
```

Comment:
`IF` uses `ELSE e2=exp_nest`, maybe we should follow that example?

Reply:
So I think `exp_nest` can either be a block or an expression, which sounds like what we want